test/functional/promise_provider.test.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+const expect = require('chai').expect;
+
+const PromiseProvider = require('../../lib/promise_provider');
+
+const candidates = {
+  withOptions: { options: { promiseLibrary: Symbol('withOptions') } },
+  withClientOptions: { clientOptions: { promiseLibrary: Symbol('withClientOptions') } },
+  withS: { s: { promiseLibrary: Symbol('withS') } }
+};
+
+describe('PromiseProvider', () => {
```

Comment:
This is a test for `PromiseProvider` just testing for the reducer, ensuring it works as expected.